galsim/base.py

```diff
@@ -2181,6 +2183,103 @@ def getScaleRadius(self):
         return self.SBProfile.getScaleRadius()
 
 
+class Spergel(GSObject):
+    """A class describing a Spergel profile.
+
+    The Spergel surface brightness profile is characterized by three properties: its Spergel index
+    `nu`, its `flux`, and either the `half_light_radius` or `scale_radius`.  Given these properties,
+    the surface brightness profile scales as I(r) ~ r^{nu} * K_{nu}(r), where K_{nu} is the
+    modified Bessel function of the second kind.
+
+    The Spergel profile is intended as a generic galaxy profile, somewhat like a Sersic profile, but
+    with the advantage of being analytic in both real space and Fourier space.  The Spergel index
+    `nu` plays a similar role to the Sersic index `n`, in that it adjusts the relative peakiness of
+    the profile core and the relative prominence of the profile wings.  At `nu = 0.5`, the Spergel
+    profile is equivalent to an Exponential profile (or alternatively an `n = 1.0` Sersic profile).
+    At `nu = -0.6` (and in the radial range near the half-light radius), the Spergel profile is
+    similar to a DeVaucouleurs profile or `n = 4.0` Sersic profile.
+
+    Note that for `nu <= 0.0`, the Spergel profile surface brightness diverges at the origin.  This
+    may lead to rendering problems if the profile is not convolved by either a PSF or a pixel and
+    the profile center is precisely on a pixel center.
+
+    Due to its analytic Fourier transform and depending on the indices `n` and `nu`, the Spergel
+    profile can be considerably faster to draw than the roughly equivalent Sersic profile.  For
+    example, the `nu = -0.6` Spergel profile is roughly 3x faster to draw than an `n = 4.0` Sersic
+    profile once the Sersic profile cache has been set up.  However, if not taking advantage of
+    the cache, for example, if drawing Sersic profiles with `n` continuously varying near 4.0 and
+    Spergel profiles with `nu` continuously varying near -0.6, then the Spergel profiles are about
+    50x faster to draw.  At the other end of the galaxy profile spectrum, the `nu = 0.5` Spergel
+    profile, `n = 1.0` Sersic profile, and the Exponential profile all take about the same amount
+    of time to draw if cached, and the Spergel profile is about 2x faster than the Sersic profile
+    if uncached.
+
+    For more information, refer to
+
+        D. N. Spergel, "ANALYTICAL GALAXY PROFILES FOR PHOTOMETRIC AND LENSING ANALYSIS,"
+        ASTROPHYS J SUPPL S 191(1), 58-65 (2010) [doi:10.1088/0067-0049/191/1/58].
+
+    Initialization
+    --------------
+
+    The allowed range of values for the `nu` parameter is -0.85 <= nu <= 100.  An exception will be
+    thrown if you provide a value outside that range.  The lower limit is set above the theoretical
+    lower limit of -1 due to numerical difficulties integrating the *very* peaky nu < -0.85
+    profiles.  The upper limit ...
```

Comment:
>Also, you forgot to finish this sentence.

So I did.  I added:
```The upper limit is set to avoid numerical difficulties evaluating the modified
    Bessel function of the second kind.
```